### DISCUSSION TITLE
Detailed differences to Jarulf und Why Black Death?

### DISCUSSION THREAD

**Discussion by toastmartin** [diasurgical/DevilutionX#6859] [Q&A] [answered]
Hello,

I just noticed the project and find it great, i love how fluent it plays, and how original it feels. I just have two questions:

- Is there a *_**detailed**_* list of changes comared to the original game (referenced by Jarulfs Guide)? I am asking because I like to min/max and there are alot of important details regarding for example number overflow of armor values of hell/nightmare monsters. Or the excact functionality of the "of thieves" suffix (does it still count against ALL damage, that ist not physical melee?)

- Why on earth did you add an option to disable hero destroying shrines but left the hero destroying Black Death-Zombies in the game? They encourage players to start new games just not to damage their character, that is a clear design mistake in diablo/HF. Starting new games just to evade a certain monster is not fun! If you want to keep them creepy, you could have the "reduced max HP"-effect removed, when a new game is started. I think then, the most players would take the challenge and figth the Black Death!

Thanks!
Toast

**Comment by AJenbo** [comment, marked as answer]
https://github.com/diasurgical/devilutionX/blob/master/docs/CHANGELOG.md

Because on earth it's not easy to make that change and still keep the game in sync. Don't like them; make a mod (very easy in 1.6.0).

**Comment by toastmartin** [comment]
Thanks for your answer. The detailed list ist very nice.
The log regarding DevilutionX 1.2.0 contains the line "Corrected behaviour of Thieves, Speed/Haste, Balance/Stability/Harmony, and Piercing/Bashing/Puncturing" Do you know wht exactly that means? The latter were prevented from stacking, right? But what changed with Thieves?

The behavoiur of the Black death was very unexpected to me, because I read here https://www.lurkerlounge.com/forums/thread-15760.html?highlight=DevilutionX that this project "will offer as many or more bug fixes & QOL improvements as my patch-mod ever did". So I assumed, that the HP loss was fixed here too. Also, I do not understand the technical difference between a mod and a recreation of the source code (probalby because I dont know much about coding).

**Reply by AJenbo** [reply]
I think thieves also reduced all damage from any type of missile

**Reply by AJenbo** [reply]
Black Death dealing damage to max HP clearly isn't a bug but an intended behavior so we have no intent of changing it. Others may create a mod on top of DevilutionX where they besides the bugfixes also change behaviours if so desired.

**Reply by toastmartin** [reply]
I do not see the difference between a Fascinating Shrine and the Black Death. Both were seemingly intended by the original game designers but both are bad game design choices.
Since you added an option to disable hurtful shrines, it would be coherent to add an option to disable the Black Death effect. Can I vote for that somewhere? :-)

**Reply by AJenbo** [reply]
Players can control if they press a shrine or not, but not if a mosters hits them or not, so that can't be done as a local option without causing desync.

But you can maybe do the same as the person that implemented that option and offer it in exchange for a major feature.

**Reply by AJenbo** [reply]
Again, in 1.6.0 you will be able to simply disable the monster. You can already do so in the beta.

**Reply by toastmartin** [reply]
When you talk about "local option" and "desync", do you refer to multiplayer games?
What do you maen with "offer it in exchange for a major feature"?

Nice to hear, that they can be disabled in the next version, that is a fix!

**Reply by AJenbo** [reply]
yes, wanting it for single player only is not going to make it simpler.
The guy that implemented disabling shrines also implemented multiplayer so that was very persuasive.

**Reply by DakkJaniels** [reply]
> I do not see the difference between a Fascinating Shrine and the Black Death. Both were seemingly intended by the original game designers but both are bad game design choices. Since you added an option to disable hurtful shrines, it would be coherent to add an option to disable the Black Death effect. Can I vote for that somewhere? :-)

The difference also is that the shrine still spawns, you just can't click on it.

> The behavoiur of the Black death was very unexpected to me, because I read here https://www.lurkerlounge.com/forums/thread-15760.html?highlight=DevilutionX that this project "will offer as many or more bug fixes & QOL improvements as my patch-mod ever did".

I don't think this person had anything to do with DevilutionX, so what it says there doesn't matter.

**Reply by AJenbo** [reply]
Well we do offer *more*, so technically what it says is true.

**Reply by DakkJaniels** [reply]
True, I guess that doesn't mean it is necessarily going to be the same ones 😂

**Reply by julealgon** [reply]
I don't understand why people keep referring to Black Death's life draining effect as a bug that needs "to be fixed".... where did they get that impression from? _How_ would something so specific like that be a bug in the first place LOL.

Oh well....

**Reply by DakkJaniels** [reply]
to be fair, I think changing it was in the context of QoL, not bug fixing. Not allowing bad shrines to be clicked isn't a bug either.

**Reply by toastmartin** [reply]
For me, the Black Death Max HP Malus is (if not a bug) a clear design flaw, because it contradicts replayability: The player fantasy is that you build a strong character during your journey. The levels you gain and the items you find will help you against the upcoming monsters. This is the core concept of the game.
When you have effects in the game that make your character permanently weaker like the bad shrines or the Black Death, you hurt that core concept in the sense, that the player is also accumulating curses on his yourney. Curses that he will never get rid of. That does not feel motivating. If feels like you should avoid these monsters entirely. So people will restart games just because they saw a yellow zombie. And restarting the game (for certain monsters/quests) is the opposite of playing the game.
Restarting is already much to prevalent to buy certain magical items. It should not be used to avoid monsters.

**Reply by StephenCWills** [reply]
> The log regarding DevilutionX 1.2.0 contains the line "Corrected behaviour of Thieves, Speed/Haste, Balance/Stability/Harmony, and Piercing/Bashing/Puncturing" Do you know wht exactly that means? The latter were prevented from stacking, right?

No, they were not prevented from stacking. However, I don't really know the details.

**Reply by julealgon** [reply]
@toastmartin nothing of what you just said is _factual_ though, it's all your opinion. I could similarly share my own opinion here, in that I believe the game was originally meant to be a rogue-like (the ability to save the game was added much later in development as confirmed in interviews: it was "always hardcode mode" before). I also strongly believe "difficulty levels" were also added much later and never thought out fully (as you can obviously tell by the fact that they don't exist normally in single player (only through a bug), and the obvious terrible balance work in multiplayer.

Replayability doesn't mean "you replay the game with the same character indefinitely", necessarily. Yes, that also is related to replayability, but the original core intent of the game (in my opinion, based on some evidence etc) was for you to replay the game _with new heros_ each time, from scratch.

The Black Death life draining mechanic goes very well with that notion, as do the negative shrines.

**Reply by toastmartin** [reply]
@julealgon You are correct. When I grew up, I saw the game as I decribed above and of course I had no clue, that there were other visions of games :-) So, when I want to play Diablo/HF today for nostalgia, I try to resemble that notion, although it has probobly been shaped by other games in the meantime.
Do you remember the time when there were no fanpages/forums to spoil the game? When the documentation was slim/english and you had to EXPLORE the game. Not only the contet but also the rules! That were the times .. 

Do you still have a link to that interview?

**Reply by DakkJaniels** [reply]
This is probably one of them:
https://www.youtube.com/watch?v=LI_CXw4x61E

Here's the original design document:
https://www.graybeardgames.com/download/diablo_pitch.pdf

**Reply by toastmartin** [reply]
:-) The video and the "design document" are really interesting. The background that David Brevik describes clearly came from tabletop RPG's and Rougelikes. On the other hand they intentionaly went away from that with the real time action property.
The replayability seemed to be important from the start. But the storys David tells seems to suport your view that the replayability is referring to the game, not the single character. Although "character development" was mentioned too .. but the balance on nightmare/hell is so bad, that seemed to be limited to completing the story on normal once.

Maybe I should widen my vision of the game .. 

@julealgon and all the other folks here: How are you playing this version of Diablo? Multiplayer? Singleplayer? Fresh Hero for each session? Are you restricting yourself somehow: No shopping in town? No Saving? Only playing full games clearing alle levels?

The other thing that feels "so long ago" about that video: Imagine! They are finishing the game, and then they are SHIPPING it. Like .. with a ship! Afterwards, they can't change it any more. No bug fixes, no cheat protection, no balance tweaks. Games were much smaller back then. :-)

**Reply by julealgon** [reply]
> @julealgon and all the other folks here: How are you playing this version of Diablo? Multiplayer? Singleplayer? Fresh Hero for each session? Are you restricting yourself somehow: No shopping in town? No Saving? Only playing full games clearing alle levels?

Been a while since I've actually played through the game, but the original is a bit too easy normally for me now, so I always play with some sort of self-imposed challenge in mind, usually Ironman. One of the last playthroughs I attempted was an axe-wielding melee Sorcerer, for instance, and I do play in "hardcore" mode, meaning death = restart. No duping, no mana-shield bug, no exploiting of vendors (save/loading, entering/exiting town, etc to refresh items, in case of the Sorcerer run).

As much as possible, I also try to avoid exploiting the AI too much with the whole visibility bug where you can hit monsters with a bow but they can't hit you back (4-tiling, etc). I just don't see the point in these exploits, it doesn't make the game more fun to me (on the contrary).

> The other thing that feels "so long ago" about that video: Imagine! They are finishing the game, and then they are SHIPPING it. Like .. with a ship! Afterwards, they can't change it any more. No bug fixes, no cheat protection, no balance tweaks. Games were much smaller back then. :-)

Yeah... games these days are released in completely broken states and fixed later (including day-1 patches and all that). For the most part (very few exceptions, usually Nintendo games).

**Reply by HuskyMACE** [reply]
I agree that Black Deaths and crippling shrines are not "bugs" that need "fixed". That being said, if a "change" were to be implemented regarding these, I wonder if it would be easier to give Pepin and/or Adria some kind of "Remove Curses" option (akin to the removed "Receive Healing"), where your maximum Life/Mana are recalculated based on your other stats. Such an option would mean that "curses" are present and meaningful in challenging contexts (namely Ironman or other self-imposed limitations), but are not irreversibly character crippling in more normal or casual circumstances. Further, purists can ignore the option as with many of the other QoL changes, and there could be one less toggle in the options menu.

I understand this suggestion may be moot and/or naive (and frankly unimportant), but just thought I would put it out there.
